server/jetstream_versioning.go

```diff
@@ -45,13 +51,29 @@ func setStreamAssetVersionMetadata(cfg *StreamConfig, prevCfg *StreamConfig) {
 	cfg.Metadata[JSRequiredLevelMetadataKey] = requiredApiLevel
 }
 
+// includeDynamicStreamAssetVersionMetadata adds dynamic fields into the (copied) metadata.
+func includeDynamicStreamAssetVersionMetadata(cfg StreamConfig) StreamConfig {
+	newCfg := cfg
```

Comment:
Why are we making another copy here? We already have a copy since we are passing by value on the stack.

Reply:
These should be addr/ptr IMO. If we need to copy that is fine but we are doing 3 copies here, one inbound, one with newCfg, and one returning the result.

Reply:
The resulting type in `StreamInfo.Config` is a `StreamConfig`, and the input is `mset.config()` that returns a `StreamConfig` as well. So it was mostly for convenience that those call sites didn't need to be adjusted apart from the function call.

Good point about the copy due to by value, have removed the use of `newCfg` causing an additional copy.
Let me know if it should still be adjusted to use pointers instead.